presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -313,7 +314,7 @@ private PlanNodeStatsEstimate calculateJoinComplementStats(
 
         // TODO: use range methods when they have defined (and consistent) semantics
         double leftNDV = leftColumnStats.getDistinctValuesCount();
-        double matchingRightNDV = rightColumnStats.getDistinctValuesCount() * unmatchedJoinComplementNdvsCoefficient;
+        double matchingRightNDV = roundToDouble(rightColumnStats.getDistinctValuesCount() * unmatchedJoinComplementNdvsCoefficient);
```

Comment:
> maybe call it round()

It will be confusing, since there's already `round()` method in `Math`, that rounds to `long`.

> roundDouble()

I chose `roundToDouble` to mimic other methods from `DoubleMath` in guava, e.g.: `roundToInt`, `roundToBigInteger`, `roundToLong`, etc. But i agree that `roundToDouble` sound confusing. 

I can rename it to `roundToNearestMathematicalIngeterDouble`, but that sound oververbose.

Reply:
> 1.5 is a perfect average expected value. Rounding it to 2 introduces 33% error.

There are no 1.5 distinct values in real world. There are either 2 or 1. Same as for the rows count. It makes it way easier to reason about the cost model when integer values are integers. In real world we cannot have 1.5 rows returned.

 I had to introduce that round to make `leftNDV <= matchingRightNDV` check work in some of the corner cases covered by the unit tests (as value returned by SymbolStatsEstimate#getDistinctValuesCount is always rounded now).